include/eve/module/core/regular/impl/swap_adjacent_groups.hpp

```diff
@@ -42,7 +42,7 @@ template<simd_value Wide, std::ptrdiff_t G>
 EVE_FORCEINLINE auto
 swap_adjacent_groups_(EVE_SUPPORTS(cpu_), Wide v, fixed<G>) noexcept requires(G <= Wide::size())
 {
-  if constexpr( G == Wide::size() ) { return v; }
+  if constexpr( G == Wide::size() || G == 0) { return v; }
```

Comment:
This should have a test

Reply:
You are right This should have a test and I added one just now. It produces a segfault !
I have investigated to see why but I don't understand.
I first modified check because I have seen a % 2*G in it, but that corrected nothing
Perhaps the problem depends of the is_swag thing, but I do not want to go further inside shuffle.
So I leave it to you. You can choose every solution that will satisfy you
1) close the pull request or tell me to close it
2) just keep the doc
3) find out why 0 behaves so badly
4) do anything else